src/sas/qtgui/Perspectives/Fitting/FittingPerspective.py

```diff
@@ -118,32 +119,21 @@ def onExcelCopy(self):
     def onLatexCopy(self):
         self.currentTab.onCopyToClipboard("Latex")
 
-    def serializeAllFitpage(self):
+    def serializeAll(self, all_data):
+        return self.serializeAllFitpage(all_data)
+
+    def serializeAllFitpage(self, all_data):
         # serialize all active fitpages and return
         # a dictionary: {data_id: fitpage_state}
-        params = {}
         for i, tab in enumerate(self.tabs):
-            tab_data = self.getSerializedFitpage(tab)
-            if 'data_id' not in tab_data: continue
-            id = tab_data['data_id'][0]
-            if isinstance(id, list):
-                for i in id:
-                    if i in params:
-                        params[i].append(tab_data)
-                    else:
-                        params[i] = [tab_data]
-            else:
-                if id in params:
-                    params[id].append(tab_data)
-                else:
-                    params[id] = [tab_data]
-        return params
+            all_data = self.getSerializedFitpage(tab, all_data)
```

Comment:
This is a bad way of calling a method. You both modify the passed argument and send it back as return value.
Wouldn't it make sense to just pass `tab`? `getSerializedFitpage` doesn't need to know the whole `all_data` content, only the tab name it needs to serialize. The result can be then just appended to `all_param` in the caller.